test/testscript.go

```diff
@@ -17,9 +17,14 @@ type TestscriptTest struct {
 	params     testscript.Params
 	setupFuncs []func(*testscript.Env) error
 	projectDir string
+	saveDir    string
 }
 
-func NewTestscriptTest(t *testing.T, projectDir string) *TestscriptTest {
+func NewTestscriptTest(
+	t *testing.T,
+	projectDir string,
+	saveDir string,
```

Comment:
this seems to strongly impose a concept on the testscript test famework that is not used by most examples. I'd be interested in seeing if we can either achieve this on a case-by-case basis at a lower level, or go the other way and come up with another higher level test setup that defines input/output relationship dependencies, eg `NewIOCoupledTest` that gives structure to the expectations about what sort of dependencies are expected to be provided at each step by the previous step

Reply:
We could then feed it an entire test function per step or something like that